engine/apps/slack/models/slack_message.py

```diff
@@ -72,26 +102,21 @@ class Meta:
 
     @property
     def slack_team_identity(self):
-        if self._slack_team_identity is None:
-            if self.organization is None:  # strange case when organization is None
-                logger.warning(
-                    f"SlackMessage (pk: {self.pk}) fields _slack_team_identity and organization is None. "
-                    f"It is strange!"
-                )
-                return None
-            self._slack_team_identity = self.organization.slack_team_identity
-            self.save()
-        return self._slack_team_identity
+        return self.channel.slack_team_identity
+
+    @property
+    def channel_slack_id(self) -> str:
+        return self.channel.slack_id
 
     @property
     def permalink(self) -> typing.Optional[str]:
-        # Don't send request for permalink if there is no slack_team_identity or slack token has been revoked
-        if self.cached_permalink or not self.slack_team_identity or self.slack_team_identity.detected_token_revoked:
```

Comment:
I verified (in production) and don't see any records where `slack_team_identity` is null for `SlackMessage` (fwiw, I think this field should eventually be dropped from this model and we should use `slack_message.channel.slack_team_identity` instead; see comments above)